### PR TITLE
re-arrange shell.nix to add new features and improve perf

### PR DESCRIPTION
**Pull request by cleverca22**
plain `nix-shell` gives you a shell with just `io`, `nixops`,and `terraform` and no cardano or ghc
`nix-shell -A ioSelfBuild` gives you a shell with `io` and `ghcid-io` functions and a ghc that can build iohk-ops
`nix-shell -A withAuxx` gives a shell with io, auxx, and tools, no ghc

### CODE REVIEW

File: shell.nix [hunk 1]
```diff
@@ -4,15 +4,29 @@
...
+    shellHook = ''
+      function io {
+        runhaskell -iiohk iohk/iohk-ops.hs
+      }
+      function ghcid-io {
+        ${pkgs.haskellPackages.ghcid}/bin/ghcid -c "ghci -iiohk iohk/iohk-ops.hs"
+      }
+    '';
+  });
+  withAuxx = pkgs.runCommand "shell" {
```
Comment (deepfire) [inline shell.nix]: @cleverca22, so this is to be used like:
```
nix-shell -A withAuxx
```
right?

Comment (cleverca22) [reply shell.nix]: yep

File: scripts/ci.sh [hunk 2]
```diff
@@ -65,6 +65,10 @@ GENERAL_OPTIONS="--verbose --deployer 0.0.0.0"
 COMMON_OPTIONS="--topology topology-min.yaml"
 CARDANO_COMPONENTS="Nodes ${WITH_EXPLORER:+Explorer} ${WITH_REPORT_SERVER:+ReportServer}"
 
+nix-build default.nix -A cardano-sl-tools -o cardano-sl-tools
```
Comment (domenkozar) [inline scripts/ci.sh]: Hmm, why is this needed? Shouldn't the code just use nix-shell if it needs these tools?

Comment (cleverca22) [reply scripts/ci.sh]: the CI runs `iohk-ops` without a nix-shell, and was relying on the fact that it was `makeWrapper'`d to have cardano-tools in `PATH`

and the whole point of this PR is to not depend on cardano when building io

Comment (domenkozar) [reply scripts/ci.sh]: When does CI need it exactly?

**Comment by domenkozar** [comment]
OK, this just needs a docs update then: https://github.com/input-output-hk/iohk-ops#development
